services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusService.kt

```diff
@@ -0,0 +1,108 @@
+            applicationEntity.applicationIdentifier,
+            applicationEntity.applicationType,
+            hakemusDataResponseWithYhteystiedot(
+                applicationEntity.id,
+                applicationEntity.applicationData,
+                applicationEntity.yhteystiedot
+            ),
+            applicationEntity.hanke.hankeTunnus
+        )
+
+    private fun hakemusDataResponseWithYhteystiedot(
+        applicationId: Long,
+        applicationData: ApplicationData,
+        hakemusyhteystiedot: Map<ApplicationContactType, HakemusyhteystietoEntity>
+    ) =
+        when (applicationData) {
+            is CableReportApplicationData ->
+                JohtoselvitysHakemusDataResponse(
+                    applicationData.applicationType,
+                    applicationData.name,
+                    customerWithContactsResponseWithYhteystiedot(
+                        hakemusyhteystiedot,
+                        ApplicationContactType.HAKIJA
+                    )
+                        ?: throw HakemusyhteystietoNotFoundException(
+                            applicationId,
+                            ApplicationContactType.HAKIJA
+                        ),
+                    applicationData.areas,
+                    applicationData.startTime,
+                    applicationData.endTime,
+                    applicationData.pendingOnClient,
+                    applicationData.workDescription,
+                    customerWithContactsResponseWithYhteystiedot(
+                        hakemusyhteystiedot,
+                        ApplicationContactType.TYON_SUORITTAJA
+                    )
+                        ?: throw HakemusyhteystietoNotFoundException(
+                            applicationId,
+                            ApplicationContactType.TYON_SUORITTAJA
+                        ),
+                    applicationData.rockExcavation,
+                    applicationData.postalAddress,
+                    customerWithContactsResponseWithYhteystiedot(
+                        hakemusyhteystiedot,
+                        ApplicationContactType.ASIANHOITAJA
+                    ),
+                    applicationData.invoicingCustomer,
+                    applicationData.customerReference,
+                    applicationData.area,
+                    customerWithContactsResponseWithYhteystiedot(
+                        hakemusyhteystiedot,
+                        ApplicationContactType.RAKENNUTTAJA
+                    ),
+                    applicationData.constructionWork,
+                    applicationData.maintenanceWork,
+                    applicationData.emergencyWork,
+                    applicationData.propertyConnectivity
+                )
+        }
+
+    private fun customerWithContactsResponseWithYhteystiedot(
+        hakemusyhteystiedot: Map<ApplicationContactType, HakemusyhteystietoEntity>,
+        contactType: ApplicationContactType
```

Comment:
The map could be indexed before calling this function.
```suggestion
        hakemusyhteystieto: HakemusyhteystietoEntity?,
```

Reply:
Nice catch, thanks!